src/orders/Orders.ts

```diff
@@ -77,44 +76,33 @@ export class Orders {
             while (this.openOrderQueue.length > 0) {
                 const order = this.openOrderQueue.shift();
                 const contractId = order?.contract?.conId;
-                const orderId = order?.order?.orderId;
+                const permId = order?.order?.permId;
 
                 // Ignore if already Filled processed / completed
-                if (this.completedTrades.has(`${orderId}`)) {
+                if (this.completedTrades.has(`${permId}`)) {
                     continue;
                 };
 
-                // ignore cancelled orders
-                if (this.cancelledOrders.has(orderId)) {
+                // Ignore cancelled orders
+                if (this.cancelledOrders.has(permId)) {
                     continue;
                 }
 
-                // ignore pre - submitted orders but update it in openOrders
-                if (order?.orderState?.status === OrderStatus.PreSubmitted) {
-                    this.openOrders.set(contractId, order);
-                    continue;
-                };
-
-                // no logs here ------------------------------------------------
-                // no logs here ------------------------------------------------
-
                 this.logOpenOrder("processOrderQueue", order);
 
-                if (contractId) {
-                    this.openOrders.set(contractId, order);
```

Comment:
here we're registering the open order, are you able to see orders list without this line?

Reply:
Yes, it's set in the switch statement now

Reply:
PreSubmitted also set there

Reply:
you put it at `default case`, 

it's pose to be before switch

Reply:
Why it should be before switch?

Reply:
I think it's not useful to add it to open orders just so it can be deleted right after if it was filled or cancelled

Reply:
because submitted orders will not be tracked, first locally and see if it works

Reply:
> you put it at `default case`,

not only at default case. the two previous case are missing "break" so it goes through.

Reply:
> because submitted orders will not be tracked, first locally and see if it works

these are tested locally

Reply:
it's more clear to have set openOpen before the switch, because what happens to other statuses ? we remove break clause too ?

Reply:
> it's more clear to have set openOpen before the switch, because what happens to other statuses ? we remove break clause too ?

I just didn't like that it's adding them to the array every time and then deletes it instantly every time orders are updated. Waste of resources.

I removed the break from other statuses. For some reason though they are supposed to be ignored.

Reply:
> Waste of resources

heheh, it's just a list

> I removed the break from other statuses

should be good then,